File: acode/abc255/c/update.py
X, A, D, N = list(map(int, input().split()))

flag = 0

if D != 0:
    flag = (X-A)/D

if flag <= 0:
    print(abs(X-A))
    exit()
if flag >= N:
    print(abs(X-(A+D*(N-1))))
    exit()


base = X - A

diff = base % D

if diff == 0:
    print(0)
else:
    print(min(abs(diff-D), abs(diff)))

File: acode/abc255/c/update.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
  ll x, a, d, n, i;
  cin >> x >> a >> d >> n;
  if (d == 0) {i = 0;} 
  else {i = (x-a)/d;}
  auto f = [&](ll i) {
    if (i < 0) i = 0;
    if (i >= n) i = n-1;
    return a+d*i;
  };
  ll ans = abs(f(i)-x);
  ans = min(ans, abs(f(i+1)-x));
  cout << ans << endl;
  return 0;
}
